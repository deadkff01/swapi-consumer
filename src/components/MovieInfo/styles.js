import styled from 'styled-components'

export const MovieCard = styled.div`
  position: relative;
  display: block;
  margin: 30px auto;
  width: 500px;
  padding: 10px;
  border: 2px solid #888888;
  border-radius: 8px;
  background-color: rgba(136, 136, 136, 0.4);
  box-sizing: border-box;

  @media (max-width: 500px) {
    width: 300px;
  }
`
export const ImgCard = styled.div`
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  width: 220px;
  height: 344px;
  position: relative;
  margin: 10px auto;
  display: block;
`
export const Field = styled.div`
  color: #fff;
  font-size: 20px;
  padding: 0px 20px 0 20px;
`

export const BackButton = styled.button`
  position: absolute;
  top: -35px;
  left: 0px;
  color: #fff;
  font-size: 15px;
  font-weight: bold;
  background-color: transparent;
  width: 166px;
  height: 28px;
  padding-left: 33px;
  text-align: left;
  background-position-x: 0px;
  background-size: 30px;
  border: none;
  background-repeat: no-repeat;
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
`
