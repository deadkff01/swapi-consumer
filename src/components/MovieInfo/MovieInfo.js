import React from 'react'
import { Redirect, Link } from 'react-router-dom'
import moment from 'moment'
import { MovieCard, ImgCard, Field, BackButton } from './styles'
import MovieTitle from '../Common/MovieTitle'

const MovieInfo = ({ location }) => {
  if (!location.state) {
    return <Redirect data-testid="redirect-movie-card" to="/" />
  }

  const { movie } = location.state

  return (
    <MovieCard data-testid="movie-card">
      <Link to="/">
        <BackButton className="back-arrow-icon">Back to home</BackButton>
      </Link>
      <ImgCard
        data-testid="movie-image"
        className={`starwarts-movie-image-${movie.episode_id}`}
        title={movie.title}
        alt={movie.title}
      />
      <MovieTitle data-testid="movie-title">{movie.title}</MovieTitle>
      <Field>
        Director: <span data-testid="movie-director">{movie.director}</span>
      </Field>
      <Field>
        Producers: <span data-testid="movie-producers">{movie.producer}</span>
      </Field>
      <Field>
        Release date:
        <span data-testid="movie-release-date">
          {moment(movie.release_date).format('DD/MM/YYYY')}
        </span>
      </Field>
    </MovieCard>
  )
}

export default MovieInfo
