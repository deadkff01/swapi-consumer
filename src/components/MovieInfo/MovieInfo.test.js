import React from 'react'
import { render, cleanup } from '@testing-library/react'
//import '@testing-library/jest-dom/extend-expect'
import { BrowserRouter } from 'react-router-dom'
import moment from 'moment'
import MovieInfo from './MovieInfo'
import allMovies from '../../__mocks__/mockMovies'

const MovieInfoComponent = props => (
  <BrowserRouter>
    <MovieInfo {...props} />
  </BrowserRouter>
)

describe('Main component', () => {
  afterEach(cleanup)

  it('renders the component', () => {
    const mockLocation = { location: { state: { movie: allMovies[0] } } }
    const container = render(<MovieInfoComponent {...mockLocation} />)
    expect(container).toMatchSnapshot()
  })

  test('test movie properties', () => {
    const mockLocation = { location: { state: { movie: allMovies[0] } } }
    const { getByTestId } = render(<MovieInfoComponent {...mockLocation} />)
    expect(getByTestId('movie-image')).toBeDefined()
    expect(getByTestId('movie-title').textContent).toBe(allMovies[0].title)
    expect(getByTestId('movie-director').textContent).toBe(allMovies[0].director)
    expect(getByTestId('movie-producers').textContent).toBe(allMovies[0].producer)
    expect(getByTestId('movie-release-date').textContent).toBe(
      moment(allMovies[0].release_date).format('DD/MM/YYYY')
    )
  })
})
