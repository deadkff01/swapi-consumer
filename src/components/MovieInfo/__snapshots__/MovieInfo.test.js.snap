// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa cGTIuu"
        data-testid="movie-card"
      >
        <a
          href="/"
        >
          <button
            class="sc-bxivhb bTfZfQ back-arrow-icon"
          >
            Back to home
          </button>
        </a>
        <div
          alt="A New Hope"
          class="sc-bwzfXH fHgHwc starwarts-movie-image-4"
          data-testid="movie-image"
          title="A New Hope"
        />
        <h1
          class="sc-ifAKCX IWCJz"
          data-testid="movie-title"
        >
          A New Hope
        </h1>
        <div
          class="sc-htpNat jxLqNZ"
        >
          Director: 
          <span
            data-testid="movie-director"
          >
            George Lucas
          </span>
        </div>
        <div
          class="sc-htpNat jxLqNZ"
        >
          Producers: 
          <span
            data-testid="movie-producers"
          >
            Gary Kurtz, Rick McCallum
          </span>
        </div>
        <div
          class="sc-htpNat jxLqNZ"
        >
          Release date:
          <span
            data-testid="movie-release-date"
          >
            25/05/1977
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa cGTIuu"
      data-testid="movie-card"
    >
      <a
        href="/"
      >
        <button
          class="sc-bxivhb bTfZfQ back-arrow-icon"
        >
          Back to home
        </button>
      </a>
      <div
        alt="A New Hope"
        class="sc-bwzfXH fHgHwc starwarts-movie-image-4"
        data-testid="movie-image"
        title="A New Hope"
      />
      <h1
        class="sc-ifAKCX IWCJz"
        data-testid="movie-title"
      >
        A New Hope
      </h1>
      <div
        class="sc-htpNat jxLqNZ"
      >
        Director: 
        <span
          data-testid="movie-director"
        >
          George Lucas
        </span>
      </div>
      <div
        class="sc-htpNat jxLqNZ"
      >
        Producers: 
        <span
          data-testid="movie-producers"
        >
          Gary Kurtz, Rick McCallum
        </span>
      </div>
      <div
        class="sc-htpNat jxLqNZ"
      >
        Release date:
        <span
          data-testid="movie-release-date"
        >
          25/05/1977
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
